### ISSUE TITLE
No support for setting query parameters in apiV1UsersUserIdCredentialsForgotPasswordPost method

### ISSUE THREAD

**Issue by oanam** [okta/okta-sdk-java#462]
Hi,

The SDK implementation of the POST /api/v1/users/${userId}/credentials/forgot_password API (found in DefaultClient class, method apiV1UsersUserIdCredentialsForgotPasswordPost), does not support passing any query parameters. 

This API supports a boolean query parameter, "sendEmail" (default = true), which cannot be set when using the SDK.
Please refer to https://developer.okta.com/docs/reference/api/users/#forgot-password for details.

We need to call this API with the sendEmail=false option. I think the SDK should be updated to support this.

Thanks in advance,
Oana

**Comment by bdemers** [comment]
Thanks for the report @oanam !

Until we get a fix in, you can do something like this: https://github.com/okta/okta-sdk-java/#call-other-api-endpoints

Which should translate to this, (from memory, so it might not be 100% correct)

```java
ForgotPasswordResponse result = client.http()
    .post("/api/v1/users/<your-id>/credentials/forgot_password", ForgotPasswordResponse.class);
```

**Comment by oanam** [comment]
Thank you, @bdemers !

As a side note, it wouldn't hurt to change the name of the method. 
The current name, "apiV1UsersUserIdCredentialsForgotPasswordPost" isn't very easy to spot - we actually missed it the first time we looked for it.

**Comment by bdemers** [comment]
Thanks @oanam!
I thought that was a typo from your first message. We will get this fixed too.  It _should_ be `User.forgotPassword(boolean sendEmail)`

**Comment by arvindkrishnakumar-okta** [comment]
Fixed in v4.0.0
